MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,899 @@
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Change the auto-generated editor settings
+
+You can change the data editor settings by using the property of `DataFormItem` in the `GenerateDataFormItem` event when AutoGenerate is true.
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "MiddleName")
+            {
+                e.DataFormItem.IsReadOnly = true;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+### Change the autogenerated editor visibility
+
+You can change the specific editor visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Setting watermark for auto-generated editor
+
+You can display the watermark in the editor by defining the display attribute or using the `GenerateDataFormItem` event when AutoGenerate is true.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(Prompt = "Enter middle name")]
+    public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
```

Comment:
GenerateDataFormItem event when AutoGenerate is true.

For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor

remove

Reply:
removed